smart_home/include/devices/device.h

```diff
@@ -0,0 +1,24 @@
+#ifndef SMART_HOME_DEVICES_DEVICE_H_
+#define SMART_HOME_DEVICES_DEVICE_H_
+
+#include "../rooms/room.h"
+
+namespace devices {
+
+// Rule of Five
+
+class Device {
+  public:
+    virtual rooms::Room* room() = 0;
+
+  protected:
+    Device(rooms::Room* room): room_(room) {}
+
+    virtual ~Device();
```

Comment:
Constructors and Destructors shouldn't be protected but public
[leaving the comment only here so that I won't spam with comments but please make the changes everywhere]